packages/catalog-process/src/services/catalogService.ts

```diff
@@ -2954,6 +2957,47 @@ export function catalogServiceBuilder(
         throw eServiceTemplateWithoutPublishedVersion(templateId);
       }
 
+      const tenant = await retrieveTenant(
+        ctx.authData.organizationId,
+        readModelService
+      );
+
+      assertTenantKindExists(tenant);
+
+      const riskAnalysis: RiskAnalysis[] = template.riskAnalysis
+        .filter((r) =>
+          match(r.tenantKind)
+            .with(tenantKind.PA, () => tenant.kind === tenantKind.PA)
+            /**
+             * For now, GSP, PRIVATE and SCP tenants share the same risk analysis .
+             * This may change in the future.
+             */
+            .with(
+              tenantKind.GSP,
+              tenantKind.PRIVATE,
+              tenantKind.SCP,
+              () =>
+                tenant.kind === tenantKind.GSP ||
+                tenant.kind === tenantKind.PRIVATE ||
+                tenant.kind === tenantKind.SCP
+            )
+            .exhaustive()
+        )
```

Comment:
This piece of code is hard to read. If I understood, what you want to do is:
- If the tenant is PA, keep only the PA risk analyses 
- If the tenant is GSP/PRIVATE/SCP, keep all the GSP/PRIVATE/SCP risk analyses

If yes, I would refactor matching on the kind of the tenant and do something like:

```ts
const riskAnalysis: RiskAnalysis[] = template.riskAnalysis.filter((r) =>
  match(tenant.kind)
    .with(tenantKind.PA, () => r.tenantKind === tenantKind.PA)
    .with(
      tenantKind.GSP,
      tenantKind.PRIVATE,
      tenantKind.SCP,
      () => [tenantKind.GSP, tenantKind.PRIVATE, tenantKind.SCP].includes(r.tenantKind)
             /**
             * For now, GSP, PRIVATE, and SCP tenants share the same risk analysis.
             * This may change in the future.
             */
    )
    .exhaustive()
);
```

Reply:
Your solution gives a type error in the includes parameter.

<img width="904" alt="Screenshot 2025-03-11 alle 15 53 28" src="https://github.com/user-attachments/assets/d1b6dfbc-7967-4ee9-b091-09a4a43cc124" />

My solution is to type it as the match function argument:
```typescript
const riskAnalysis: RiskAnalysis[] = template.riskAnalysis.filter((r) =>
  match(tenant.kind)
    .with(tenantKind.PA, () => r.tenantKind === tenantKind.PA)
    .with(
      tenantKind.GSP,
      tenantKind.PRIVATE,
      tenantKind.SCP,
      (t) => [tenantKind.GSP, tenantKind.PRIVATE, tenantKind.SCP].includes(r.tenantKind as typeof t)
             /**
             * For now, GSP, PRIVATE, and SCP tenants share the same risk analysis.
             * This may change in the future.
             */
    )
    .exhaustive()
);
```

Reply:
Ok, you could try to bring back the comparison as before, but keeping the main match on the tenant.kind:

```suggestion
const riskAnalysis: RiskAnalysis[] = template.riskAnalysis.filter((r) =>
  match(tenant.kind)
    .with(tenantKind.PA, () => r.tenantKind === tenantKind.PA)
    .with(
      tenantKind.GSP,
      tenantKind.PRIVATE,
      tenantKind.SCP,
      () =>  r.tenantKind === tenantKind.GSP || r.tenantKind === tenantKind.PRIVATE || r.tenantKind === tenantKind.SCP
             /**
             * For now, GSP, PRIVATE, and SCP tenants share the same risk analysis.
             * This may change in the future.
             */
    )
    .exhaustive()
);
```

Or maybe also something like the following, but it's not so readable...

```ts
 match<[TenantKind, TenantKind]>([tenant.kind, r.tenantKind])
    .with([tenantKind.PA, tenantKind.PA], () => true)
    .with([tenantKind.PA, P.not(tenantKind.PA)], () => false)
    .with([P.not(tenantKind.PA), tenantKind.PA], () => false)
    .with([P.not(tenantKind.PA), P.not(tenantKind.PA)], () => true)
    .exhaustive()
```